tutorial/consuming_packages/cross_building_with_conan.rst

```diff
@@ -0,0 +1,217 @@
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that we have our two profiles prepared, let's have a look at our *conanfile.py*:
+
+.. code-block:: python
+    :caption: **conanfile.py**
+
+    from conan import ConanFile
+    from conan.tools.cmake import cmake_layout
+
+    class CompressorRecipe(ConanFile):
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "CMakeToolchain", "CMakeDeps"
+
+        def requirements(self):
+            self.requires("zlib/1.2.11")
+            self.tool_requires("cmake/3.19.8")
+
+        def layout(self):
+            cmake_layout(self)
+
+As you can see, this is practically the same *conanfile.py* we used in the previous
+example. We will require **zlib/1.2.11** as a regular dependency and **cmake/3.19.8** as a
+tool needed for building the application. Also, we are using the pre-defined
+``cmake_layout``.
+
+We will need the application to build for the Raspberry Pi with the cross-build
+toolchain and also link the **zlib/1.2.11** library built for the same platform. On the
+other side, we need the **cmake/3.19.8** binary to run in Ubuntu Linux. Conan manages this
+internally in the dependency graph differentiating between what we call the "build
+context" and the "host context":
+
+* The **host context** is populated with the root package (the one specified in the
+  :command:`conan install` or :command:`conan create` command) and all its requirements.
+  In some cases, it may also include tools that should run in the host machine (for example
+  a test framework that will run in the host machine). In this case, this includes the
+  compressor application and the **zlib/1.2.11** dependency.
+
+* The **build context** contains the tool requirements used in the build machine. This
+  category typically includes all the developer tools like CMake, compilers and linkers.
+  In this case, this includes the **cmake/3.19.8** tool.
+
+
+These contexts define how Conan will manage each one of the dependencies. For example, as
+**zlib/1.2.11** belongs to the **host context**, the ``[buildenv]`` build environment we
+defined in the **raspberry** profile (profile host) will only apply to the **zlib/1.2.11**
+library when building and won't affect anything that belongs to the **build context** like
+the **cmake/3.19.8** dependency.
+
+Now, let's build the application. First, call :command:`conan install` with the
+profiles for the build and host platforms. This will install the  **zlib/1.2.11**
+dependency built for *armv7hf* architecture and a **cmake/3.19.8** version that runs for
+64-bit architecture.
+
+.. code-block:: bash
+    
+    $ conan install . --build missing -pr:b=./profiles/ubuntu -pr:h=./profiles/raspberry
+
+Then, let's call CMake to build the application. As we did in the previous example we have
+to activate the **build environment** running ``source generators/conanbuild.sh``. That will
+set the environment variables needed to locate the cross-build toolchain and build the
```

Comment:
I am missing here information about this build environment file. I understand the purpose but would like to understand how conan is able to generate that file for me and where the information comes from.
- Does it come from the profile?
- Is Conan detecting something for me? 
- Does it work "out of the box" or do I have to configure something?

Maybe those questions arise because I have not read (or can't remember) other/previous sections of the documentation. If it is too much, then a link to that reference could be useful for the reader. But I think this "build environment" setup is key for users doing crossbuilding.

Reply:
There's a brief introduction [here](https://docs.conan.io/en/2.0-alpha/tutorial/consuming_packages/use_tools_as_conan_packages.html) about the VirtualBuildEnv generator and also here to the [VirtualRunEnv](https://docs.conan.io/en/2.0-alpha/tutorial/consuming_packages/different_configurations.html). I'll add a comment there so that the user is aware that adding the [buildenv] section the VirtualBuildEnv generator is automatically invoked.